File: changelog_builder/commit_processor/unique_issues.py
import re


class UniqueIssues:
    def __init__(self, config):
        self.config = config

    def process(self, commits):
        issues = {}
        valid_commits = {}

        pattern = self.config['issue_format']
        for key in set(commits):
            match = re.search(pattern, commits[key])
            if None == match:
                valid_commits[key] = commits[key]
            else:
                issues[match.group(0)] = key

        for key in issues:
            revision = issues[key]
            valid_commits[revision] = commits[revision]

        return valid_commits

File: lib/processor/uniqueissues.js
function UniqueIssues() {

}

UniqueIssues.prototype.process = function (config, commits) {
	if (!config.unique_issues) return commits;

	var issues = {};
	var validCommits = {};
	var pattern = config.issue_format;

	for (var key in commits) {
		if (commits.hasOwnProperty(key) && commits[key]) {
			var commit = commits[key];
			var match = commit.match(pattern);
			if (!match) {
				validCommits[key] = commit;
			} else {
				issues[match[0]] = key;
			}
		}
	}

	for (key in issues) {
		var revision = issues[key];
		validCommits[revision] = commits[revision];
	}

	return validCommits;
};

module.exports = UniqueIssues;
